lib/inferno/apps/cli/evaluate.rb

```diff
@@ -1,14 +1,29 @@
 require_relative '../../../inferno/dsl/fhir_evaluation/evaluator'
+require_relative '../../../inferno/dsl/ig'
+require_relative '../../utils/ig_downloader'
+
+require 'tempfile'
 
 module Inferno
   module CLI
-    class Evaluate
-      def run(ig_path, data_path, _log_level)
+    class Evaluate < Thor::Group
+      include Thor::Actions
+      include Inferno::Utils::IgDownloader
+
+      def evaluate(ig_path, data_path, _log_level)
         validate_args(ig_path, data_path)
 
-        # IG Import, rule execution, and result output below will be integrated at phase 2 and 3.
+        if File.file?(ig_path)
+          ig = Inferno::DSL::IG.from_file(ig_path)
+        else
+          Tempfile.create('package.tgz') do |temp_file|
```

Comment:
If wrong file path (e.g. `bundle exec inferno evaluate ./spec/fixtures/uscore311.tgzzz`), I think it should stop and error before going to IG Downloader.

Reply:
In principle I agree with this, but I'm not sure how to implement it. Technically an IG identifier like `hl7.fhir.us.core@3.1.1` is also potentially a file path, so we'd have to have some kind of logic to determine whether the string entered was *intended* to be a filepath or not, which gets tricky. Any thoughts?

Reply:
You're right. If file path is in the form of identifier, there's no way to know it from expression unless actually try and see if it fails. In this code we will know it if it fails twice, one by file path and the other by identifier. It won't be a huge issue in terms of performance since we only import one IG per execution. Also as Steve mentioned, later we may consider start from list of fhir pkgs in the system to see whether there is IG or not. That will naturally solve this.